src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -1452,13 +1452,19 @@ XChainCreateBridge::preclaim(PreclaimContext const& ctx)
 {
     auto const account = ctx.tx[sfAccount];
     auto const bridgeSpec = ctx.tx[sfXChainBridge];
-
     STXChainBridge::ChainType const chainType =
         STXChainBridge::srcChain(account == bridgeSpec.lockingChainDoor());
 
-    if (ctx.view.read(keylet::bridge(bridgeSpec, chainType)))
     {
-        return tecDUPLICATE;
+        auto hasBridge = [&](STXChainBridge::ChainType ct) -> bool {
+            return !!ctx.view.read(keylet::bridge(bridgeSpec, ct));
```

Comment:
As a small optimization, consider calling `ctx.view.exists()` instead, like this:
```
return ctx.view.exists(keylet::bridge(bridgeSpec, ct));
```
This will avoid deserializing the bridge to check the `LedgerEntryType`.  Just a thought.

Reply:
Love it. Fixed in `8bbf7b9e21 [fold] Use exists instead of read to check if ledger object exists`